### PR TITLE
Fix const static

### PR DESCRIPTION
**Pull request by zdtsw**
compilation error:  error: ‘static’ is not at beginning of declaration [-Werror=old-style-declaration]

### CODE REVIEW

**Comment by zdtsw** [comment]
sry, made two commits on the same PR......
they are actually for two different issues

**Comment by travisspencer** [comment]
More good fixes. Thanks, @zdtsw !

**Comment by travisspencer** [comment]
Tested this here like this:

`./configure --without-pcre --without-http_rewrite_module --without-http_gzip_module --prefix=$HOME/nginx-dev --with-cc-opt=-Werror=old-style-declaration --with-cc=/usr/bin/gcc`

Got this error:

```
/usr/bin/gcc -c -pipe  -O -Wall -Wextra -Wpointer-arith -Wconditional-uninitialized -Wno-unused-parameter -Wno-deprecated-declarations -Werror -g -Werror=old-style-declaration -I src/core -I src/event -I src/event/modules -I src/os/unix -I objs \
		-o objs/src/core/nginx.o \
		src/core/nginx.c
error: unknown warning option '-Werror=old-style-declaration'; did you mean '-Werror=out-of-line-declaration'? [-Werror,-Wunknown-warning-option]
make[2]: *** [objs/src/core/nginx.o] Error 1
make[1]: *** [build] Error 2
make: *** [default] Error 2
```

My `gcc` is:

```
gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

I think this is the default `gcc` on Mac. Is this flag a newer thing or?

**Comment by zdtsw** [comment]
I had it run on both gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4) and gcc version 4.8.5 20150623 (Red Hat 4.8.5-11) (GCC), did not get any error.

**Comment by zdtsw** [comment]
But apart from this, I do have one more problem to compile nginx source.....
src/core/ngx_string.h:103:69: error: value computed is not used [-Werror=unused-value]
 #define ngx_cpymem(dst, src, n)   (((u_char *) memcpy(dst, src, n)) + (n))
but this is not related to these two commits i made

**Comment by zdtsw** [comment]
/usr/bin/gcc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -Werror=old-style-declaration -I src/core -I src/event -I src/event/modules -I src/os/unix -I objs -I src/http -I src/http/modules. I guess my problem was from "-Wno-unused-parameter" , just need to suspend warning which treats it as error

File: phantom_token.c [hunk 1]
```diff
@@ -15,6 +15,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#define _GNU_SOURCE
```
Comment (travisspencer) [inline phantom_token.c]: is this needed? We're not using any GNU/Linux extensions in here that I'm aware of.

Comment (travisspencer) [reply phantom_token.c]: ah, `strcasestr`. I guess we should see about switching to `ngx_strcasestrn` and skip this GNU extra.

**Comment by zdtsw** [comment]
just made a new commit with updates

File: phantom_token.c [hunk 2]
```diff
@@ -247,7 +246,7 @@ static ngx_int_t handler(ngx_http_request_t *request)
 
     u_char *bearer_token_pos;
 
-    if ((bearer_token_pos = (u_char *)strcasestr((char*)request->headers_in.authorization->value.data, BEARER)) == NULL)
+    if ((bearer_token_pos = (u_char *)ngx_strcasestrn((char*)request->headers_in.authorization->value.data, BEARER,BEARER_SIZE)) == NULL)
```
Comment (travisspencer) [inline phantom_token.c]: The `u_char *` cast is unnecessary with that function, it seems. On Mac, I get an error if `BEARER` isn't cast to `char*` from `const char*`. Therefore, I suggest this: 

`if ((bearer_token_pos = ngx_strcasestrn(request->headers_in.authorization->value.data, (char*)BEARER, BEARER_SIZE)) == NULL)`

**Comment by zdtsw** [comment]
yep! also did two type casting

File: phantom_token.c [hunk 3]
```diff
@@ -246,7 +246,7 @@ static ngx_int_t handler(ngx_http_request_t *request)
 
     u_char *bearer_token_pos;
 
-    if ((bearer_token_pos = (u_char *)ngx_strcasestrn((char*)request->headers_in.authorization->value.data, BEARER,BEARER_SIZE)) == NULL)
+    if ((bearer_token_pos = ngx_strcasestrn((u_char*)request->headers_in.authorization->value.data,(char*) BEARER,BEARER_SIZE)) == NULL)
```
Comment (travisspencer) [inline phantom_token.c]: the `u_char*` doesn't hurt anything I suppose, but is it needed?

```
 ngx_str_t

        A structure containing a pointer to a character array and the length of the character array

    size_t len

        The length of the character array

    unsigned char *data
```
https://www.nginx.com/resources/wiki/extending/api/utility/

Sorry to nitpick.

Comment (zdtsw) [reply phantom_token.c]: Got some "note" from compiler, saying: 
src/core/ngx_string.h:163:9: note: expected ‘char *’ but argument is of type ‘const char *’
src/core/ngx_string.h:163:9: note: expected ‘u_char * {aka unsigned char *}’ but argument is of type ‘char *’
. feel annoying to have these, that's why did the casting.

Comment (zdtsw) [reply phantom_token.c]: one thing I forgot to mention before, did uplift for nginx source from 1.13.3 to 1.13.5 locally

Comment (travisspencer) [reply phantom_token.c]: Must just be a difference between your compiler on Linux vs. mine on Mac. It's fine to have the cast and avoid the warning (which I agree is annoying).

**Pull request merged**